File: scala-galaxy/src/main/java/com/galaxy/scala/config/XxlConfig.scala
package com.galaxy.scala.config

import org.springframework.beans.factory.annotation.Value
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.context.annotation.{Configuration, PropertySource}

/**
  * Created by wangpeng on 19/03/2018. 
  */

@Configuration
@PropertySource(value = Array("classpath:xxl.properties"))
class XxlConfig {

  @Value("${xxl.job.admin.addresses}")
  private var addresses: String = _

  @Value("${xxl.job.executor.appname}")
  private var appname: String = _

  @Value("${xxl.job.executor.host}")
  private var host: String = _

  @Value("${xxl.job.executor.logpath}")
  private var logpath: String = _

  @Value("${xxl.job.accessToken}")
  private var accessToken: String = _

  @Value("${xxl.job.executor.port}")
  private var port: Integer = _
}

File: simple-galaxy/src/main/java/com/galaxy/config/XXLConfig.java
package com.galaxy.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * Created by wangpeng on 2018/4/3.
 */
@Configuration
@ConfigurationProperties(prefix="xxl.job")
public class XXLConfig {

    @Value("${xxl.job.db.driverClass}")
    private String DB_DRIVER;

    @Value("${xxl.job.db.url}")
    private String DB_URL;

    @Value("${xxl.job.db.user}")
    private String DB_USERNAME;

    @Value("${xxl.job.db.password}")
    private String DB_PASSWORD;

    @Value("${xxl.job.accessToken}")
    private String ACCESS_TOKEN;

    @Override
    public String toString() {
        return "XXLConfig{" +
                "DB_DRIVER='" + DB_DRIVER + '\'' +
                ", DB_URL='" + DB_URL + '\'' +
                ", DB_USERNAME='" + DB_USERNAME + '\'' +
                ", DB_PASSWORD='" + DB_PASSWORD + '\'' +
                ", ACCESS_TOKEN='" + ACCESS_TOKEN + '\'' +
                '}';
    }

    @Bean
    public XXLConfig init(){
        System.out.println(toString());
        return new XXLConfig();
    }
}
